node/src/main/scala/coop/rchain/node/configuration/model.scala

```diff
@@ -2,66 +2,85 @@ package coop.rchain.node.configuration
 
 import java.nio.file.Path
 
-import scala.concurrent.duration.FiniteDuration
+import com.typesafe.config.Config
 
+import scala.concurrent.duration.FiniteDuration
 import coop.rchain.casper.util.comm.ListenAtName.Name
 import coop.rchain.comm.PeerNode
 import coop.rchain.crypto.{PrivateKey, PublicKey}
+import coop.rchain.casper.CasperConf
+import coop.rchain.comm.transport.TlsConf
+import coop.rchain.node.configuration.Configuration.Profile
+import pureconfig._
+import pureconfig.generic.auto._
+
+final case class NodeConf(
+    standalone: Boolean,
+    protocolServer: ProtocolServer,
+    protocolClient: ProtocolClient,
+    peersDiscovery: PeersDiscovery,
+    apiServer: ApiServer,
+    tls: TlsConf,
+    storage: Storage,
+    casper: CasperConf,
+    metrics: Metrics,
+    // This field is dynamic and computer according to profile and is not used in client code
+    // But it is required in the model to enable strict configuration check for unknown keys
+    defaultDataDir: String
```

Comment:
How this "strict check" works? Who is doing the checking and when?

Reply:
https://github.com/rchain/rchain/pull/2912/commits/7b1015b6b360d2d044f73a9546b773fb82a1d712

Reply:
What will happen if you remove this field? I'm not sure how can someone understand where is the connection to behavior you mentioned in the comment.

Reply:
Looks like I should to improve wordings. 
https://github.com/rchain/rchain/blob/38596efe3b9fb79a8651bf2bf1a0999337094e9f/node/src/main/resources/defaults.conf#L130 Here is the default hocon file, it should be a le to be loaded to NodeConf case lass. This `default-data-dir` is initialized from profile provided through CLI and added to default config. So its in HOCON file. And therefore it should be in model as well, or when trying to create model Pureconfig an error will be thrown `unknown key`. And we need these unknown key to throw errors to make sure mistypes in config file are detected.

Reply:
Adjusted comment in https://github.com/rchain/rchain/pull/2912/commits/afd3bcf0b7dd03a75190849048411d2fe21fa736

Reply:
Thanks for additional config and explanation.